worker/src/main/java/gov/cms/ab2d/worker/config/JobMessageSource.java

```diff
@@ -0,0 +1,29 @@
+package gov.cms.ab2d.worker.config;
+
+import org.springframework.integration.jdbc.JdbcPollingChannelAdapter;
+
+import javax.sql.DataSource;
+
+/**
+ * Extends the JdbcPollingChannelAdapter to implement a Job table specific polling Adapter.
+ * The SQL query is specified as a static constant.
+ */
+public class JobMessageSource extends JdbcPollingChannelAdapter {
+
+    private static final String QUERY_GET_NEXT_UNPROCESSED_JOB = "                  " +
+                                    "    SELECT id, job_uuid, status                " +
+                                    "      FROM job                                 " +
+                                    "     WHERE status = 'SUBMITTED'                " +
+                                    "       AND (SELECT count(lock_key)             " +
+                                    "              FROM int_lock                    " +
+                                    "             WHERE lock_key = job_uuid) = 0    " +
+                                    "  ORDER BY created_at;                         ";
+
```

Comment:
I was wondering, if I should also add a `LIMIT 1` at the end of this query.
I did not add it because line 25 below has a `setMaxRows(1);` 

Thoughts?

Reply:
@spathiyil I think we are good as is. I don't feel strongly either way though